From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Charlotte Hartmann Paludo <git@charlotteharludo.com>
Date: Wed, 5 Nov 2025 13:01:25 +0100
Subject: [PATCH] runtime: assign GPU devices to multiple containers

Signed-off-by: Charlotte Hartmann Paludo <git@charlotteharludo.com>
---
 src/runtime/virtcontainers/container.go | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/src/runtime/virtcontainers/container.go b/src/runtime/virtcontainers/container.go
index 2140eb106f133932a83e716fa937bfb3419e4f58..0112f40b77056dcbaa57dd8d7b925e5f7fd3a32f 100644
--- a/src/runtime/virtcontainers/container.go
+++ b/src/runtime/virtcontainers/container.go
@@ -11,6 +11,7 @@ import (
 	"io"
 	"os"
 	"path/filepath"
+	"slices"
 	"sort"
 	"strconv"
 	"strings"
@@ -1120,13 +1121,19 @@ func (c *Container) annotateContainerWithVFIOMetadata(devices interface{}) {
 
 		}
 
+		// If the NVIDIA_VISIBLE_DEVICES envvar exists and is set to "all" for the given container,
+		// we attach ALL Nvidia GPU devices to this container.
+		if slices.Contains(c.config.CustomSpec.Process.Env, "NVIDIA_VISIBLE_DEVICES=all") {
+			c.siblingAnnotation("", siblings)
+		}
 	}
 }
+
 func (c *Container) siblingAnnotation(devPath string, siblings []DeviceRelation) {
 	for _, sibling := range siblings {
-		if sibling.Path == devPath {
+		if sibling.Path == devPath || slices.Contains(c.config.CustomSpec.Process.Env, "NVIDIA_VISIBLE_DEVICES=all") {
 			// We have here either /dev/vfio/<num> or /dev/vfio/devices/vfio<num>
-			baseName := filepath.Base(devPath)
+			baseName := filepath.Base(sibling.Path)
 			vfioNum := baseName
 			// For IOMMUFD format /dev/vfio/devices/vfio<num>, strip "vfio" prefix
 			if strings.HasPrefix(baseName, "vfio") {
