From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: davidweisse <98460960+davidweisse@users.noreply.github.com>
Date: Mon, 18 May 2026 12:57:48 +0200
Subject: [PATCH] genpolicy: support pod level resource limits

---
 src/tools/genpolicy/src/pod.rs | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/tools/genpolicy/src/pod.rs b/src/tools/genpolicy/src/pod.rs
index 4964ff6eab320e5914ccde8f7ddfe3ac2909da39..7d8548576a87946bebcce8369771014e956cd3a9 100644
--- a/src/tools/genpolicy/src/pod.rs
+++ b/src/tools/genpolicy/src/pod.rs
@@ -107,6 +107,9 @@ pub struct PodSpec {
 
     #[serde(skip_serializing_if = "Option::is_none")]
     schedulerName: Option<String>,
+
+    #[serde(skip_serializing_if = "Option::is_none")]
+    resources: Option<ResourceRequirements>,
 }
 
 /// See Reference / Kubernetes API / Workload Resources / Pod.
