diff -Naur a/fs/Kconfig b/fs/Kconfig
--- a/fs/Kconfig	2023-11-30 22:09:55.000000000 +0100
+++ b/fs/Kconfig	2023-11-30 22:09:55.000000000 +0100
@@ -23,6 +23,7 @@
 source "fs/ext2/Kconfig"
 source "fs/ext4/Kconfig"
 source "fs/jbd2/Kconfig"
+source "fs/tarfs/Kconfig"
 
 config FS_MBCACHE
 # Meta block cache for Extended Attributes (ext2/ext3/ext4)
diff -Naur a/fs/Makefile b/fs/Makefile
--- a/fs/Makefile	2023-11-30 22:09:55.000000000 +0100
+++ b/fs/Makefile	2023-11-30 22:09:55.000000000 +0100
@@ -76,6 +76,7 @@
 obj-$(CONFIG_JBD2)		+= jbd2/
 obj-$(CONFIG_CRAMFS)		+= cramfs/
 obj-$(CONFIG_SQUASHFS)		+= squashfs/
+obj-$(CONFIG_TARFS_FS)		+= tarfs/
 obj-y				+= ramfs/
 obj-$(CONFIG_HUGETLBFS)		+= hugetlbfs/
 obj-$(CONFIG_CODA_FS)		+= coda/
