From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Meyer <49727155+katexochen@users.noreply.github.com>
Date: Thu, 11 Jul 2024 12:05:00 +0200
Subject: [PATCH] genpolicy: regex check contrast specific layer-src-prefix

Signed-off-by: Paul Meyer <49727155+katexochen@users.noreply.github.com>
---
 src/tools/genpolicy/rules.rego | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/genpolicy/rules.rego b/src/tools/genpolicy/rules.rego
index aa0488ae2dffc63780967ca706ea2c8ffcfb1391..38171ac667d6084b085633aedf669ed381c10e22 100644
--- a/src/tools/genpolicy/rules.rego
+++ b/src/tools/genpolicy/rules.rego
@@ -944,7 +944,7 @@ allow_storage_options(p_storage, i_storage, layer_ids, root_hashes) {
     i_count == p_count + 3
 
     print("allow_storage_options 2: i_storage.options[0] =", i_storage.options[0])
-    i_storage.options[0] == "io.katacontainers.fs-opt.layer-src-prefix=/var/lib/containerd/io.containerd.snapshotter.v1.tardev/layers"
+    regex.match(`io\.katacontainers\.fs-opt\.layer-src-prefix=/var/lib/containerd/io\.containerd\.snapshotter\.v1\.tardev-contrast-cc-(aks|k3s|rke2)-(qemu|clh)-(snp|tdx)-[a-f0-9]{8}/layers`, i_storage.options[0])
 
     print("allow_storage_options 2: i_storage.options[i_count - 2] =", i_storage.options[i_count - 2])
     i_storage.options[i_count - 2] == "io.katacontainers.fs-opt.overlay-rw"
