From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jmxnzo <jakob.lammering@ruhr-uni-bochum.de>
Date: Fri, 10 Jan 2025 14:49:05 +0100
Subject: [PATCH] genpolicy: support mount propagation and ro-mounts

---
 src/tools/genpolicy/rules.rego | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/genpolicy/rules.rego b/src/tools/genpolicy/rules.rego
index 38171ac667d6084b085633aedf669ed381c10e22..f2cd235360dfd81df48650cd6c89c20a690025ad 100644
--- a/src/tools/genpolicy/rules.rego
+++ b/src/tools/genpolicy/rules.rego
@@ -109,7 +109,7 @@ allow_create_container_input {
     count(i_linux.GIDMappings) == 0
     count(i_linux.MountLabel) == 0
     count(i_linux.Resources.Devices) == 0
-    count(i_linux.RootfsPropagation) == 0
+    # TODO(burgerdev): is it harmful to always allow RootfsPropagation?
     count(i_linux.UIDMappings) == 0
     is_null(i_linux.IntelRdt)
     is_null(i_linux.Resources.BlockIO)
