From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: jmxnzo <jakob.lammering@ruhr-uni-bochum.de>
Date: Fri, 10 Jan 2025 14:49:05 +0100
Subject: [PATCH] genpolicy: support mount propagation and ro-mounts

---
 src/tools/genpolicy/rules.rego | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/tools/genpolicy/rules.rego b/src/tools/genpolicy/rules.rego
index 1b92071d1f28f40fbbd17f3a78705d690c42ac82..c16439f24e2e591d115bc1d54ee1d9fb085cd775 100644
--- a/src/tools/genpolicy/rules.rego
+++ b/src/tools/genpolicy/rules.rego
@@ -306,7 +306,7 @@ allow_create_container_input(req) {
     count(i_linux.GIDMappings) == 0
     count(i_linux.MountLabel) == 0
     count(i_linux.Resources.Devices) == 0
-    count(i_linux.RootfsPropagation) == 0
+    # TODO(burgerdev): is it harmful to always allow RootfsPropagation?
     count(i_linux.UIDMappings) == 0
     is_null(i_linux.IntelRdt)
     is_null(i_linux.Resources.BlockIO)
