diff --git a/genpolicy-rules.rego b/genpolicy-rules.rego
index 823e5e76d55bac47ad9c79d8916f92702efa316d..e3acc09dd931fd8d028eadf356679a97dfa4e633 100644
--- a/genpolicy-rules.rego
+++ b/genpolicy-rules.rego
@@ -199,9 +199,9 @@ allow_by_sandbox_name(p_container, i_oci, i_storages, s_name) {
     p_namespace := p_oci.Annotations[s_namespace]
     i_namespace := i_oci.Annotations[s_namespace]
     print("allow_by_sandbox_name: p_namespace =", p_namespace, "i_namespace =", i_namespace)
-    p_namespace == i_namespace
+    regex.match("^[a-z0-9-]{1,63}$", i_namespace)

-    allow_by_container_types(p_oci, i_oci, s_name, p_namespace)
+    allow_by_container_types(p_oci, i_oci, s_name, i_namespace)
     allow_by_bundle_or_sandbox_id(p_container, i_oci, i_storages)
     allow_process(p_oci, i_oci, s_name)
