From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Markus Rudy <mr@edgeless.systems>
Date: Mon, 29 Sep 2025 18:48:13 +0200
Subject: [PATCH] agent: remove initdata processing

Signed-off-by: Markus Rudy <mr@edgeless.systems>
---
 src/agent/src/main.rs | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/src/agent/src/main.rs b/src/agent/src/main.rs
index a362396b6f29d26ce3d3b8e9ec2bf343f94cf156..3cfeeb431beccb639652ca8eee9e811bf862961d 100644
--- a/src/agent/src/main.rs
+++ b/src/agent/src/main.rs
@@ -411,33 +411,6 @@ async fn start_sandbox(
     let (tx, rx) = tokio::sync::oneshot::channel();
     sandbox.lock().await.sender = Some(tx);
 
-    let initdata_return_value = initdata::initialize_initdata(logger).await?;
-
-    let gc_procs = config.guest_components_procs;
-    let launch_plan = build_coco_launch_plan(config, &initdata_return_value, gc_procs)?;
-    if !attestation_components_available(logger, &launch_plan) {
-        warn!(
-            logger,
-            "attestation binaries requested for launch not available"
-        );
-    } else {
-        init_attestation_components(logger, &launch_plan).await?;
-    }
-
-    // if policy is given via initdata, use it
-    #[cfg(feature = "agent-policy")]
-    if let Some(initdata_return_value) = initdata_return_value {
-        if let Some(policy) = &initdata_return_value._policy {
-            info!(logger, "using policy from initdata");
-            AGENT_POLICY
-                .lock()
-                .await
-                .set_policy(policy)
-                .await
-                .context("Failed to set policy from initdata")?;
-        }
-    }
-
     let mut oma = None;
     let mut _ort = None;
     if let Some(c) = &config.mem_agent {
